### PR TITLE
Replaces book of babel tendril loot with ominous mirror

### PR DESCRIPTION
**Pull request by LazennG**
# Document the changes in your pull request

adds ominous mirror to the game, a new item that summons a ghost role helper akin to holoparasites.
-can move around within 20 spaces of the mirror 
-has 20 hp, does 14 brute melee damage to nonfauna enemies, which take 50. 
-can fire a projectile every couple seconds that does 5 damage to nonfauna and 20 to fauna
-can swap positions with whoever is holding the mirror every 15 seconds
-can mine basalt that it walks into similar to concussive gauntlets

-Upon 'dying,' the reflection is locked in the mirror for 20 seconds where they slowly heal
-When the user isn't conscious, the reflection can't leave the mirror
-the reflection can't pull living things
-the reflection can't manifest itself outside the mirror if it's not being held

The mirror holder can: 
-call the reflection back to the mirror 
-swap the reflection out for a different ghost for circumstances similar to where you'd do the same for a holopara

# Why is this good for the game?
book of babel sucks and i dont like a lot of ghost roles being unable to do anything without being an antag so i think this is nice.

# Testing
the entering/exiting of the mirror and the swap
dreamseeker_O7KokzW6GM

being banished back to the mirror after 'dying'
dreamseeker_PIu3rEJuL7

# Spriting

mirrorcrack
mirrorcrackedreal

# Wiki Documentation

https://wiki.yogstation.net/wiki/Shaft_Miner#Necropolis_Chest_Loot
replace book of babel section with:
 A mirror that summons a fragile companion that can assist in mining and fighting, as well as swap locations with the holder. When defeated, they're returned back to the mirror for 20 seconds, and they can't exit the mirror while the holder is unconscious or when the mirror isn't held.

# Changelog

:cl:  @thegoldencat413 @Chubbygummibear @LazennG 
rscadd: Added ominous mirror
rscdel: Removed book of babel from tendril loot pool 
imageadd: added ominous mirror sprites
/:cl:
 thanks goldencat for spriting this and helping me develop it a freaking year ago

### CODE REVIEW

**Review by Mqiib — APPROVED** [review]
I like spooky mirror sprite

**Comment by GDRdev** [comment]
you know- this seems... kinda op? sorta like a budget jaunt holo

**Comment by LazennG** [comment]
new changes: 
-the mirror buddy is no longer dense
-bullets pass through and do 1/4th of the damage to the buddy

**Comment by thegoldencat413** [comment]
why do the sprites suck? is the spriter stupid

**Comment by cowbot92** [comment]
im looking for the man in the mirror

**Comment by Chubbygummibear** [comment]
the maid in the mirror

File: yogstation/code/modules/mining/lavaland/doppelmirror.dm [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+	..()
+	original = user
+
+/obj/item/dopmirror/attack_self(mob/living/user)
+	if(reflected)
+		return
+	if(!(GLOB.ghost_role_flags & GHOSTROLE_STATION_SENTIENCE))
+		to_chat(user, span_notice("Anomalous otherworldly energies keep the mirror from reflecting anything!"))
+		return
+
```
Comment (Moltijoe) [inline yogstation/code/modules/mining/lavaland/doppelmirror.dm:28]: add a check to prevent spamming the attack_self

**Review by Moltijoe — CHANGES_REQUESTED** [review]

File: yogstation/code/modules/mining/lavaland/doppelmirror.dm (hunk 1 above)
Comment (Moltijoe) [reply yogstation/code/modules/mining/lavaland/doppelmirror.dm:28]: use th ecooldown system

**Review by Moltijoe — CHANGES_REQUESTED** [review]

File: yogstation/code/modules/mining/lavaland/doppelmirror.dm [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+	maxbodytemp = INFINITY
+	atmos_requirements = list("min_oxy" = 0, "max_oxy" = 0, "min_tox" = 0, "max_tox" = 0, "min_co2" = 0, "max_co2" = 0, "min_n2" = 0, "max_n2" = 0)
+	movement_type = FLYING
+	var/obj/item/dopmirror/mirror = null// the mirror that's returned to on dying
+	var/melee_fauna_bonus = 36
+	var/hibernating = FALSE
+	var/move_range = 20
+	var/datum/action/innate/jumpback/jumpback
+	var/datum/action/innate/appear/appear
+	var/datum/action/innate/swap/swap
```
Comment (Moltijoe) [inline yogstation/code/modules/mining/lavaland/doppelmirror.dm:154]: make sure to delete these when the mob is destroyed

**Review by Moltijoe — CHANGES_REQUESTED** [review]

File: yogstation/code/modules/mining/lavaland/doppelmirror.dm [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+		playsound(src, 'sound/effects/glassbr2.ogg', 75)
+		src.update_icon()
+	else
+		to_chat(user, "... and your reflection stares back at you. Try again later.")
+
+/obj/item/dopmirror/Destroy()
+	to_chat(reflected, span_userdanger("Your world shatters."))
+	reflected.jumpback.Remove(reflected)
+	reflected.appear.Remove(reflected)
+	reflected.swap.Remove(reflected)
```
Comment (Moltijoe) [inline yogstation/code/modules/mining/lavaland/doppelmirror.dm]: put these instead on the destroy proc of the mob itself
and make sure to qdel them after removing

**Review by Moltijoe — CHANGES_REQUESTED** [review]

**Review by Moltijoe — APPROVED** [review]

**Pull request merged**